### PR TITLE
cleanup error_handler related code

### PR DESCRIPTION
**Pull request by zhichao-cao**
Remove code not in use, add comments, remove redundant code.

Test plan: make check

### CODE REVIEW

File: db/error_handler_fs_test.cc [hunk 1]
```diff
@@ -1874,6 +1877,8 @@ TEST_F(DBErrorHandlingFSTest, FLushWritNoWALRetryableErrorAutoRecover2) {
   Destroy(options);
 }
 
+// Auto resume fromt he flush retryable IO error. Activate the FS before the
```
Comment (anand1976) [inline db/error_handler_fs_test.cc:1880]: "fromt he" -> "from the". Likewise with the following two comments.

File: db/error_handler.cc [hunk 2]
```diff
@@ -406,10 +406,7 @@ const Status& ErrorHandler::SetBGError(const IOStatus& bg_io_err,
     // it can directly overwrite any existing bg_error_.
     bool auto_recovery = false;
     Status bg_err(new_bg_io_err, Status::Severity::kUnrecoverableError);
-    bg_error_ = bg_err;
-    if (recovery_in_prog_ && recovery_error_.ok()) {
-      recovery_error_ = bg_err;
-    }
+    CheckAndSetRecoveryAndBGError(bg_err);
```
Comment (anand1976) [inline db/error_handler.cc:409]: Do we know why the original code wasn't checking bg_error_ severity before setting it?

**Review by anand1976 — APPROVED** [review]
LGTM

File: db/error_handler_fs_test.cc (hunk 1 above)
Comment (zhichao-cao) [reply db/error_handler_fs_test.cc:1880]: Thanks for the review! Will change accordingly.

File: db/error_handler.cc (hunk 2 above)
Comment (zhichao-cao) [reply db/error_handler.cc:409]: The io_error is classified as"Status::Severity::kUnrecoverableError", which is the worst one. It is safe to directly override bg_error_.

**Pull request closed without merging**